server/setup_flow.go

```diff
@@ -271,44 +289,56 @@ func (p *Plugin) stepServerConfigureAppLink2() flow.Step {
-func (p *Plugin) stepCloudEnableDeveloperMode() flow.Step {
-	return flow.NewStep(stepCloudEnableDeveloperMode).
-		WithPretext("##### :white_check_mark: Step 2: Configure the Mattermost app in Jira").
-		WithTitle("Enable development mode.").
-		WithText("Integrating Mattermost with Jira Cloud requires setting your Jira instance to development mode (see _screenshot_). " +
-			"Enabling development mode allows you to install apps like Mattermost from outside the Atlassian Marketplace.\n" +
-			"Complete the following steps in Jira, then come back here to select **Continue**.\n\n" +
-			"1. Navigate to [**Settings > Apps > Manage Apps**]({{.JiraURL}}/plugins/servlet/upm?source=side_nav_manage_addons).\n" +
-			"2. Select **Settings** at the bottom of the page.\n" +
-			"3. Select **Enable development mode**, then select **Apply**.\n").
-		WithImage("public/cloud-enable-dev-mode.png").
-		OnRender(p.trackSetupWizard("setup_wizard_jira_config_start", map[string]interface{}{
-			keyEdition: CloudInstanceType,
-		})).
-		WithButton(continueButton(stepCloudUploadApp)).
-		WithButton(cancelButton)
-}
-
-func (p *Plugin) stepCloudUploadApp() flow.Step {
-	return flow.NewStep(stepCloudUploadApp).
-		WithTitle("Upload the Mattermost app to Jira.").
-		WithText("To finish the configuration, create a new app in your Jira instance.\n" +
-			"Complete the following steps, then come back here to select **Continue**.\n\n" +
-			"1. From [**Settings > Apps > Manage Apps**]({{.JiraURL}}/plugins/servlet/upm?source=side_nav_manage_addons) select **Upload app** (see _screenshot_).\n" +
-			"2. In the **From this URL field**, enter: `{{.ACURL}}` [link]({{.ACURL}}), then select **Upload**.\n" +
-			"3. Wait for the app to install. Once completed, you should see an \"Installed and ready to go!\" message.\n").
-		WithImage("public/cloud-upload-app.png").
+func (p *Plugin) stepCloudOAuthConfigure() flow.Step {
+	return flow.NewStep(stepCloudOAuthConfigure).
+		WithPretext("##### :white_check_mark: Step 2: Register an OAuth 2.0 Application in Jira").
+		WithText(fmt.Sprintf("Complete the following steps, then come back here to select **Configure**.\n\n"+
+			"1. Create an OAuth 2.0 application in Jira from the [Developer console](https://developer.atlassian.com/console/myapps/create-3lo-app/).\n"+
+			"2. Name your app according to its purpose, for example:  `Mattermost Jira Plugin - <your company name>`.\n"+
+			"3. Accept the **Terms** and click **Create**.\n"+
+			"4. Select **Permissions** in the left menu. Next to the JIRA API, select **Add**.\n"+
+			"5. Then select **Configure** and ensure following scopes are selected:\n"+
+			"       %s\n"+
+			"6. Select **Authorization** in the left menu.\n"+
+			"7. Next to OAuth 2.0 (3LO), select **Add** and set the Callback URL as follows and click **Save Changes**:\n"+
+			"       {{.OAuthCompleteURL}}\n"+
+			"8. Select **Settings** in the left menu.\n"+
+			"9. Copy the **Client ID** and **Secret** and keep it handy.\n"+
```

Comment:
check if this is `Client Secret` or just `Secret`

Reply:
Its `Secret`